api/src/test/java/jakarta/data/metamodel/restrict/TextRestrictionRecordTest.java

```diff
@@ -31,72 +32,64 @@ class TextRestrictionRecordTest {
+//            soft.assertThat(restriction.isEscaped()).isFalse();
         });
     }
 
     @Test
     void shouldCreateTextRestrictionWithExplicitNegation() {
         TextRestrictionRecord<String> restriction = new TextRestrictionRecord<>(
                 "title",
-                Operator.NOT_LIKE,
-                "%Java%"
+                Like.pattern("%Java%"),
+                true
         );
 
         SoftAssertions.assertSoftly(soft -> {
             soft.assertThat(restriction.attribute()).isEqualTo("title");
             soft.assertThat(restriction.comparison()).isEqualTo(Operator.NOT_LIKE);
-            soft.assertThat(restriction.value()).isEqualTo("%Java%");
-            soft.assertThat(restriction.isCaseSensitive()).isTrue();
-            soft.assertThat(restriction.isEscaped()).isFalse();
+            soft.assertThat(restriction.constraint().string()).isEqualTo("%Java%");
+            soft.assertThat(restriction.constraint().caseSensitive()).isTrue();
+//            soft.assertThat(restriction.isEscaped()).isFalse();
         });
     }
 
     @Test
     void shouldIgnoreCaseForTextRestriction() {
-        TextRestrictionRecord<String> restriction = new TextRestrictionRecord<>(
+        TextRestrictionRecord<String> caseInsensitiveRestriction = new TextRestrictionRecord<>(
                 "title",
-                Operator.LIKE,
-                "%Java%"
+                Like.pattern("%Java%").ignoreCase()
         );
 
-        Restriction<String> caseInsensitiveRestriction = restriction.ignoreCase();
-
         SoftAssertions.assertSoftly(soft -> {
-            soft.assertThat(caseInsensitiveRestriction).isInstanceOf(TextRestrictionRecord.class);
-            TextRestrictionRecord<String> textRestriction = (TextRestrictionRecord<String>) caseInsensitiveRestriction;
-            soft.assertThat(textRestriction.attribute()).isEqualTo("title");
-            soft.assertThat(textRestriction.comparison()).isEqualTo(Operator.LIKE);
-            soft.assertThat(textRestriction.value()).isEqualTo("%Java%");
-            soft.assertThat(textRestriction.isCaseSensitive()).isFalse();
-            soft.assertThat(textRestriction.isEscaped()).isFalse();
+            soft.assertThat(caseInsensitiveRestriction.attribute()).isEqualTo("title");
+            soft.assertThat(caseInsensitiveRestriction.comparison()).isEqualTo(Operator.LIKE);
+            soft.assertThat(caseInsensitiveRestriction.constraint().string()).isEqualTo("%Java%");
+            soft.assertThat(caseInsensitiveRestriction.constraint().caseSensitive()).isFalse();
+//            soft.assertThat(textRestriction.isEscaped()).isFalse();
         });
     }
 
     @Test
     void shouldCreateTextRestrictionWithEscapedValue() {
         TextRestrictionRecord<String> restriction = new TextRestrictionRecord<>(
                 "title",
-                Operator.LIKE,
-                true,
-                "%Java%"
+                Like.pattern("%Java%")
```

Comment:
adding escape character that can be asserted
```suggestion
                Like.pattern("%Java%", '_', '%', '^')
```

Reply:
I added a new test